worker/src/metrics.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+use std::sync::Once;
+
+#[derive(Clone)]
+pub struct Metrics {
+    pub worker_metrics: Option<WorkerMetrics>,
+}
+
+static mut METRICS: Metrics = Metrics {
+    worker_metrics: None,
+};
+static INIT: Once = Once::new();
+
+/// Initialises the metrics. Should be called only once when the worker
+/// node is initialised, otherwise it will lead to erroneously creating
+/// multiple registries.
+pub fn initialise_metrics(metrics_registry: &Registry) -> Metrics {
+    unsafe {
+        INIT.call_once(|| {
+            // Essential/core metrics across the worker node
+            let node_metrics = WorkerMetrics::new(metrics_registry);
+
+            METRICS = Metrics {
+                worker_metrics: Some(node_metrics),
+            }
+        });
+        METRICS.clone()
+    }
+}
+
+#[derive(Clone)]
```

Comment:
Because this is used inside the [Option](https://github.com/MystenLabs/narwhal/pull/426/files#diff-82e6ef919a8598919323ec7291bb8146821c64739d319b4ae7b12864fd534a96R8) in Metrics we need to have clone implemented